frontends/ol-utilities/src/learning-resources/learning-resources.ts

```diff
@@ -116,12 +120,59 @@ const formatRunDate = (
   return null
 }
 
+/**
+ * Checks if all runs of a given learning resource are identical in terms of price, delivery method, and location.
+ *
+ * @param resource - The learning resource to check.
+ * @returns `true` if all runs have the same price, delivery method, and location; otherwise, `false`.
+ */
+const allRunsAreIdentical = (resource: LearningResource) => {
+  if (!resource.runs) {
+    return true
+  }
+  if (resource.runs.length <= 1) {
+    return true
+  }
+  const prices: LearningResourcePrice[] = []
+  const deliveryMethods = []
+  const locations = []
+  for (const run of resource.runs) {
+    if (run.resource_prices) {
+      run.resource_prices.forEach((price) => {
+        if (price.amount !== "0") {
```

Comment:
Is it intentional that several resource runs all with `"0"` price are considered different?

Reply:
I tweaked this logic a bit. The idea here is that a free course that offers a certificate is always going to have the "0" price along with one other, so the course doesn't have just "one" price. I could just check `resource.free` in the conditional at the bottom, but instead I'm checking it here and just ignoring the zero price if the resource is free. That way, if there are any different prices for the certificate they will be the only ones in the set.